pkg/ir/xds.go

```diff
@@ -9,6 +29,19 @@ type Xds struct {
 	HTTP []*HTTPListener
 }
 
+// Validate the fields within the Xds structure
+func (x *Xds) Validate() error {
+	if x.Name == "" {
+		return ErrXdsNameEmpty
+	}
+	for _, http := range x.HTTP {
+		if err := http.Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
```

Comment:
Would it be worth returning *all* the validation errors, rather than just the first one encountered?

Reply:
sure can do